### PR TITLE
Feature/car controls

### PR DESCRIPTION
**Pull request by m4rtint**

### CODE REVIEW

File: Assets/Prefabs/DebugFloor.mat [hunk 1]
```diff
@@ -0,0 +1,77 @@
+%YAML 1.1
+%TAG !u! tag:unity3d.com,2011:
+--- !u!21 &2100000
+Material:
+  serializedVersion: 6
```
Comment (mikeslgoh) [inline Assets/Prefabs/DebugFloor.mat]: What is this currently used for?

File: Assets/Script/Game/Car/CarController.cs [hunk 2]
```diff
@@ -87,104 +77,200 @@ public void SetIndicatorTarget(Transform target)
...
+        foreach (WheelCollider wheel in m_wheels)
+        {
+            wheel.motorTorque = 0.0001f;
+        }
     }
 
     private void OnDestroy()
     {
-        BaseWorldManager.OnFixedUpdate -= CarControllerOnUpdate;
+        //BaseWorldManager.OnFixedUpdate -= CarControllerOnUpdate;
```
Comment (mikeslgoh) [inline Assets/Script/Game/Car/CarController.cs]: Is there a reason why you commented this out?

File: Assets/Script/Game/Car/CarController.cs [hunk 3]
```diff
@@ -87,104 +77,200 @@ public void SetIndicatorTarget(Transform target)
...
+        // Checks if the Horn Button is Pressed
+        AngryGooseGames.InputManager.IsHornPressed(PlayerInformation.PlayerControllerInformation);
+
+        m_throttle = AngryGooseGames.InputManager.MainVertical(PlayerInformation.PlayerControllerInformation);
+        float horizontal = AngryGooseGames.InputManager.MainHorizontal(PlayerInformation.PlayerControllerInformation);
+
+        bool isBreaking = AngryGooseGames.InputManager.IsBreaking(PlayerInformation.PlayerControllerInformation);
+        float brake = isBreaking ? 1 : 0;
+
+
```
Comment (mikeslgoh) [inline Assets/Script/Game/Car/CarController.cs]: is it possible to reduce the space in between lines for this function?

File: Assets/Script/Game/Car/CarController.cs [hunk 4]
```diff
@@ -87,104 +77,200 @@ public void SetIndicatorTarget(Transform target)
...
-        m_rigidbody2D.angularVelocity = AngryGooseGames.InputManager.MainHorizontal(PlayerInformation.PlayerControllerInformation) * m_currentTorqueForce * -1;
+        // Downforce
+        m_rigidbody.AddForce(-transform.up * m_carProperties.Speed * m_carProperties.Downforce);
+    }
 
-        m_currentDriftFactor = m_driftFactorSticky;
-        
-        if (RightVelocity().magnitude > m_maxStickyVelocity)
+    /*
+    private void Jump()
```
Comment (mikeslgoh) [inline Assets/Script/Game/Car/CarController.cs]: Is this code from someone else as well? Should we just remove it for now until we decide whether we want it rather have it just commented out?

File: Assets/Prefabs/DebugFloor.mat (hunk 1 above)
Comment (m4rtint) [reply Assets/Prefabs/DebugFloor.mat]: the 8 map is only roads, no floor. this is for the floor

File: Assets/Script/Game/Car/CarController.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: The map needs to have the BaseWorldManager to do anything
right now the BaseWorldManager is broken - createCamera, Create PLayer...all broken .

File: Assets/Script/Game/Car/CarController.cs (hunk 3 above)
Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: yuh, will do

File: Assets/Script/Game/Car/CarController.cs (hunk 4 above)
Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: Yes it's not used anywhere, i'm just not sure if we want 
1)Boost
2)Jump

Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: I'll remove it

File: Assets/Script/Game/Car/CarController.cs (hunk 2 above)
Comment (mikeslgoh) [reply Assets/Script/Game/Car/CarController.cs]: ah right. You did mention that last time we talked

File: Assets/Script/Game/Car/CarController.cs (hunk 4 above)
Comment (mikeslgoh) [reply Assets/Script/Game/Car/CarController.cs]: I think BOOST might be discussion we have when we talk about powerups. Jumping I'm not sure whether we want players to be able to do that (unless it's also another powerup idea?)

Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: well it's a feature, and right now (i'm not 100% sure how), it auto reflips itself upright

Comment (mikeslgoh) [reply Assets/Script/Game/Car/CarController.cs]: Oh it reflips itself? That might be good to use for ramps then rather than a jumping mechanic?

Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: either works, could have both.. like mario kart, can jump while flying off the ramp.

Comment (mikeslgoh) [reply Assets/Script/Game/Car/CarController.cs]: Yeah let's have this as a discussion topic for Wednesday's meeting then or have this noted somewhere

**Comment by mikeslgoh** [comment]
[Car controls in 3D](https://trello.com/c/2Lb1H776/282-car-controls-in-3d)

File: Assets/Script/Game/Car/CarController.cs [hunk 5]
```diff
@@ -87,104 +77,146 @@ public void SetIndicatorTarget(Transform target)
...
+        if (m_rigidbody != null && m_centerOfMass != null)
+        {
+            m_rigidbody.centerOfMass = m_centerOfMass.localPosition;
+        }
+
+        // Set the motor torque to a non null value because 0 means the wheels won't turn no matter what
+        foreach (WheelCollider wheel in m_wheels)
+        {
+            wheel.motorTorque = 0.0001f;
+        }
```
Comment (Gursaanj) [inline Assets/Script/Game/Car/CarController.cs]: Seems a bit hacky, but I guess there is no other way huh?? (That is simple at least_

Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: how is it hacky?

Comment (m4rtint) [reply Assets/Script/Game/Car/CarController.cs]: This is just how the 3rd party library works.

**Pull request merged**